### PR TITLE
refactor: Migrates CloudBackUpRestoreJobs resource to new Atlas SDK

### PR DESCRIPTION
**Pull request by lantoli**
## Proposed changes

_Jira ticket:_ [INTMDB-1080](https://jira.mongodb.org/browse/INTMDB-1080)

Migrating CloudBackUpRestoreJobs resource to new Atlas SDK.

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)
- [ ] This change requires a documentation update

## Manual QA performed:

- [ ] cfn invoke for each of CRUDL/cfn test
- [ ] Updated resource in  [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples)
- [x] Published to AWS private registry
- [x] Used the template in [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples) to create and update a stack in AWS
- [x] Deleted stack to ensure resources are deleted
- [ ] Created multiple resources in same stack
- [x] Validated in Atlas UI
- [x] Included screenshots

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code
- [ ] For CFN Resources: I have released by changes in the private registry and proved by change
  works in Atlas

## Further comments

CreatedAt is removed as it's not being sent by the server, e.g. read operation before this PR:

```
 cfn invoke --function-name TestEntrypoint resource READ  test/cloud-backup-restore-job.sample-cfn-request.json                        <aws:mms-scratch> <region:us-west-2>
Resource schema is valid.
=== Handler input ===
{
  "callbackContext": null,
  "action": "READ",
  "requestData": {
    "resourceProperties": {
      "ProjectId": "***",
      "InstanceName": "***",
      "InstanceType": "cluster",
      "SnapshotId": "***",
      "Id": "***",
      "Profile": "default"
    },
    "previousResourceProperties": {}
  },
  "region": "us-east-1",
  "awsAccountId": "***",
  "bearerToken": "***"
}
=== Handler response ===
{
  "message": "Read complete",
  "status": "SUCCESS",
  "resourceModel": {
    "ProjectId": "***",
    "InstanceType": "cluster",
    "InstanceName": "***",
    "Id": "***",
    "DeliveryType": "download",
    "Cancelled": "false",
    "Expired": "false",
    "Timestamp": "2023-10-17T09:26:33Z",
    "SnapshotId": "***",
    "Links": [
      {
        "Rel": "self",
        "Href": "***"
      },
      {
        "Rel": "***",
        "Href": "***"
      },
      {
        "Rel": "***",
        "Href": "***"
      },
      {
        "Rel": "***",
        "Href": "***"
      }
    ],
    "Profile": "default",
    "SynchronousCreationOptions": {}
  },
  "bearerToken": "***",
  "resourceModels": null
}
```

**Contract testing:**
```
cfn test --function-name TestEntrypoint --verbose   
7 passed, 3 skipped, 14 deselected
```

**Manual testing creating and deleting the stack using private registry:**

stack

atlas_create

atlas_delete

### CODE REVIEW

**Comment by Adelmar92** [comment]
@lantoli please dont close this pull request before this one https://github.com/mongodb/mongodbatlas-cloudformation-resources/pull/661

File: cfn-resources/cloud-backup-restore-jobs/cmd/resource/resource.go [hunk 1]
```diff
@@ -456,78 +349,89 @@ func isTimeOutReached(startTime string, timeOutInSeconds int) bool {
...
+	model.DeliveryType = &job.DeliveryType
+	model.ExpiresAt = util.TimePtrToStringPtr(job.ExpiresAt)
+	model.CreatedAt = nil // not available in new SDK
+	model.Id = job.Id
+	model.FinishedAt = util.TimePtrToStringPtr(job.FinishedAt)
+	model.SnapshotId = job.SnapshotId
+	model.TargetProjectId = job.TargetGroupId
+	model.Timestamp = util.TimePtrToStringPtr(job.Timestamp)
+	model.Cancelled = job.Cancelled
+	model.Failed = job.Failed
```
[the remark this answers is not in the document]
Comment (lantoli) [reply cfn-resources/cloud-backup-restore-jobs/cmd/resource/resource.go:398]: yes, it is, it was also in old SDK but for some reason it was not exported so we needed to do some workarounds in the code

[the remark this answers is not in the document]
Comment (lantoli) [reply cfn-resources/cloud-backup-restore-jobs/cmd/resource/resource.go:396]: thanks for noticing this. Although it's in the prev SDK it's not really being sent by the server so I've deleted it, see PR description for more info.

File: cfn-resources/cloud-backup-restore-jobs/cmd/resource/resource.go [hunk 2]
```diff
@@ -456,78 +349,89 @@ func isTimeOutReached(startTime string, timeOutInSeconds int) bool {
...
+	model.Links = flattenLinks(job.Links)
 }
 
-func isJobFailed(job mongodbatlas.CloudProviderSnapshotRestoreJob) bool {
-	return job.Failed != nil && *job.Failed
+func updateModelServer(model *Model, job *admin.DiskBackupSnapshotRestoreJob) {
+	model.TargetClusterName = job.TargetClusterName
+	model.DeliveryType = &job.DeliveryType
+	model.ExpiresAt = util.TimePtrToStringPtr(job.ExpiresAt)
+	model.CreatedAt = nil // not available in new SDK
```
[the remark this answers is not in the document]
Comment (lantoli) [reply cfn-resources/cloud-backup-restore-jobs/cmd/resource/resource.go]: thanks for noticing this. Although it's in the prev SDK it's not really being sent by the server so I've deleted it, see PR description for more info.

File: cfn-resources/cloud-backup-restore-jobs/mongodb-atlas-cloudbackuprestorejobs.json [hunk 3]
```diff
@@ -158,7 +154,6 @@
     "/properties/DeliveryUrl",
     "/properties/Cancelled",
     "/properties/Failed",
-    "/properties/CreatedAt",
```
Comment (AgustinBettati) [inline cfn-resources/cloud-backup-restore-jobs/mongodb-atlas-cloudbackuprestorejobs.json:153]: Would make use of the changelog that was created for this resource to mention this removal and the reasoning. On another hand I see this attribute is present in terraform, we can create a ticket to deprecate the field there.

**Review by AgustinBettati — APPROVED** [review]

**Review by AgustinBettati — APPROVED** [review]
LGTM, thanks for addressing comments

**Pull request merged**